modules/network/src/main/java/org/apache/ignite/internal/network/direct/stream/DirectByteBufferStreamImplV1.java

```diff
@@ -236,27 +237,34 @@ public void writeBoxedByte(@Nullable Byte val) {
     /** {@inheritDoc} */
     @Override
     public void writeShort(short val) {
-        lastFinished = buf.remaining() >= 2;
+        lastFinished = buf.remaining() >= 3;
 
         if (lastFinished) {
+            val++;
```

Comment:
Should not this code be extracted somewhere, like `VarIntUtils`?

Reply:
It's already copy-pasted multiple times within this class. I think I can extract the common part into a private method, would that be good?